src/pages/dataElements/New.tsx

```diff
@@ -1,5 +1,183 @@
-import React from 'react'
+import { useDataEngine } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { NoticeBox } from '@dhis2/ui'
+import { FORM_ERROR } from 'final-form'
+import React, { useEffect, useRef } from 'react'
+import { Form } from 'react-final-form'
+import { useNavigate } from 'react-router-dom'
+import { StandardFormActions, StandardFormSection } from '../../components'
+import { SCHEMA_SECTIONS } from '../../constants'
+import { getSectionPath } from '../../lib'
+import { Attribute } from '../../types/generated'
+import { DataElementFormFields, useCustomAttributesQuery } from './form'
+import { FormValues } from './form/types'
+import classes from './New.module.css'
 
+const listPath = `/${getSectionPath(SCHEMA_SECTIONS.dataElement)}`
+
+function computeInitialValues(customAttributes: Attribute[]) {
```

Comment:
I think we can leverage `zod` for this and `formatFormValues` below. But I can look into that when looking into validation.

Reply:
> Also this looks quite similiar to form/computeInitialValues ?

Yeah, I tried to create a function that both can use, reverted my changes as the only thing that's similar is the shape of the returned object. Populating it with values is completely different, so having an "abstraction" on top of that would just increase complexity but not have any benefit.